include/cyclic_reduction.hxx

```diff
@@ -100,30 +103,33 @@ public:
   ///                where N is set in the constructor or setup
   /// @param[in] b   Diagonal values. Should have size [nsys][N]
   /// @param[in] c   Right diagonal. Should have size [nsys][N]
-  void setCoefs(int nsys, T **a, T **b, T **c) {
+  void setCoefs(int nsys, Matrix<T> &a, Matrix<T> &b, Matrix<T> &c) {
+    TRACE("CyclicReduce::setCoefs");
+    
     // Make sure correct memory arrays allocated
     allocMemory(nprocs, nsys, N);
 
     // Fill coefficient array
     for(int j=0;j<Nsys;j++)
       for(int i=0;i<N;i++) {
-        coefs[j][4*i] = a[j][i];
-        coefs[j][4*i + 1] = b[j][i];
-        coefs[j][4*i + 2] = c[j][i];
+        coefs(j, 4*i) = a(j, i);
+        coefs(j, 4*i + 1) = b(j, i);
+        coefs(j, 4*i + 2) = c(j, i);
         // 4*i + 3 will contain RHS
       }
   }
 
   /// Solve a single triadiagonal system
   /// 
-  void solve(T rhs[], T x[]) {
+  void solve(Matrix<T> &rhs, Matrix<T> &x) {
     // Solving single system
     solve(1, &rhs, &x);
   }
 
   /// Solve a set of tridiagonal systems
   /// 
-  void solve(int nrhs, T **rhs, T **x) {
+  void solve(int nrhs, Matrix<T> &rhs, Matrix<T> &x) {
```

Comment:
Can the `nrhs` argument be removed? This should be the number of columns in `rhs`? I guess it is bounds checked anyway, but shouldn't be needed.

Reply:
Now removed.